webapp/packages/core-blocks/src/Menu/Menu.tsx

```diff
@@ -95,34 +100,61 @@ export const Menu = observer<IMenuProps, HTMLButtonElement>(
     }, [menuVisible]);
 
     useLayoutEffect(() => {
-      if (!mouseContextMenu?.position) {
-        return;
-      }
+      if (contextInputRef?.current && menuVisible) {
+        const inputRect = contextInputRef.current.getBoundingClientRect();
+        const caretPosition = contextInputRef.current.selectionStart || 0;
 
-      if (menuVisible) {
-        menu.hide();
-        return;
-      }
+        const span = document.createElement('span');
+        span.style.position = 'absolute';
+        span.style.visibility = 'hidden';
+        span.style.whiteSpace = 'pre';
+        span.textContent = contextInputRef.current.value.substring(0, caretPosition);
+
+        document.body.appendChild(span);
```

Comment:
Please use the menu anchor (invisible menu trigger / button) instead (you can position it as you need)